include/ui/color_picker_button.h

```diff
@@ -0,0 +1,43 @@
+/*
+* Copyright 2017 CodiLime
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*/
+#pragma once
+
+#include <QColor>
+#include <QColorDialog>
+#include <QPushButton>
+
+namespace veles {
+namespace ui {
+
+class ColorPickerButton : public QPushButton
+{
+  Q_OBJECT
+
+ public:
+  ColorPickerButton(const QColor& initial_color, QWidget* parent = nullptr);
+  QColor getColor() const;
+
+ private slots:
```

Comment:
Future-proofing: shouldn't this be a public function, to allow changing the color from outside?

Reply:
Sure, fixed.